### DISCUSSION TITLE
Fluent option with wrong text value entered when updating data

### DISCUSSION THREAD

**Discussion by jobou363** [microsoft/fluentui-blazor#2541] [General]
I try to use fluent combo box options but i'm wondering if the component could be missing a function to handle when the option typed and entered is not found that the value is change so the error can be handled.

The problem happen when i want to modify an existing entity object i fill the data on screen and I fill a combo box with the initial enum value.

Here by example the combo box is load with the Variable value.

![image](https://github.com/user-attachments/assets/747ad6d8-6fac-411f-893e-955923a96823)

I have 3 possibles values ( null, Fixe and Variable ).

![image](https://github.com/user-attachments/assets/677e3802-24bf-421c-b5aa-abe66d0e2d09)

If the user type a wrong value like by example "sadas" and press "enter" the value behind is not changed for another value and the box stay with the green valid border.

If i press on the save button the entity is save with the initial value because it wasn't change by the user and nothing is indicating or blocking the user because the selection or data entered was wrong.

![image](https://github.com/user-attachments/assets/8de8a7ff-0f91-49b6-82cb-51e884b11719)

If i select the empty/null value the validation is called and a message is shown.
![image](https://github.com/user-attachments/assets/2b4532c3-0008-41cf-9188-5dfcab610308)

I don't have the problem when i add an entry because the value is set to null than the validation is triggered.

There is not EventCallbacks available to manage this use case.

--- 

  ` <FluentCombobox Items=@Options
                   TOption="Option<string?>"
                   OptionText="@(i => i.Text?.ToString())" 
                   OptionValue="@(i => i.Value?.ToString())" 
                   OptionDisabled="@(i => i.Disabled)"
                   OptionSelected="@(i => i.Selected)"
                   SelectedOptionChanged="@SelectedOptionChanged"
                   Height="300px" 
                   Autocomplete="ComboboxAutocomplete.Both" 
                   @attributes="@Attributes" >
   </FluentCombobox>
`

**Comment by vnbaaij** [comment]
You can't limit a Combobox to only allow typing to select from the list it shows. You can use the FluentAutoComplete for that
